src/sonic-yang-models/tests/yang_model_tests/yangTest.json

```diff
@@ -642,6 +642,63 @@
 		}
 	},
 
+	"PORT_VALID_AUTONEG_TEST": {
+		"sonic-port:sonic-port": {
+			"sonic-port:PORT": {
+				"PORT_LIST": [{
+					"port_name": "Ethernet8",
+					"alias": "eth8",
+					"lanes": "65",
+					"description": "Ethernet8",
+					"speed": 25000,
+					"fec": "rs",
+					"mtu": 9000,
+					"admin_status": "up",
+					"pfc_asym": "off",
+					"autoneg": 0
+				}]
+			}
+		}
+	},
+
+	"PORT_INVALID_AUTONEG_TEST_1": {
+		"sonic-port:sonic-port": {
+			"sonic-port:PORT": {
+				"PORT_LIST": [{
+					"port_name": "Ethernet8",
+					"alias": "eth8",
+					"lanes": "65",
+					"description": "Ethernet8",
+					"speed": 25000,
+					"fec": "rs",
+					"mtu": 9000,
+					"admin_status": "up",
+					"pfc_asym": "off",
+					"autoneg": -1
```

Comment:
Maybe this is an invalid value because the type is unsigned int. And for +ve range is checked.